ljh/ljh.go

```diff
@@ -174,6 +175,100 @@ func (w *Writer) WriteRecord(rowcount int64, timestamp int64, data []uint16) err
 	return nil
 }
 
+// Writer writes LJH3.0 files
+type Writer3 struct {
+	ChanNum         int
+	Timebase        float64
+	NumberOfRows    int
+	NumberOfColumns int
+	Row             int
+	Column          int
+	HeaderWritten   bool
+	FileName        string
+	RecordsWritten  int
+
+	file *os.File
+}
+
+type HeaderTDM struct {
+	NumberOfRows    int
+	NumberOfColumns int
+	Row             int
+	Column          int
+}
+
+type Header struct {
+	Frameperiod   float64   `json:"frameperiod"`
+	Format        string    `json:"File Format"`
+	FormatVersion string    `json:"File Format Version"`
+	TDM           HeaderTDM `json:"TDM"`
+}
+
+func (w *Writer3) WriteHeader() error {
+	if w.HeaderWritten {
+		return errors.New("header already written")
+	}
+	h := Header{Frameperiod: w.Timebase, Format: "LJH3", FormatVersion: "3.0.0",
+		TDM: HeaderTDM{NumberOfRows: w.NumberOfRows, NumberOfColumns: w.NumberOfColumns,
+			Row: w.Row, Column: w.Column}}
+	s, err := json.MarshalIndent(h, "", "    ")
+	if err != nil {
+		panic("MarshallIndent error")
+	}
+	_, err = w.file.Write(s)
```

Comment:
I think it's more idiomatic here and in several following lines to save a line and say, e.g.,

```
if _,err = w.file.Write(s); err != nil {
    return err
}
```

This only works if you don't need to save either of the 2 return value from `w.file.Write`, because they would go out of scope once the if statement ends.